### PR TITLE
Exit with failure if the upload fails for some reason

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

File: kokoro/release_publish.sh [hunk 1]
```diff
@@ -46,12 +46,6 @@ GetSessionID() {
 	eval $__resultvar="'$nxsessionid'"
 }
 
-# Usage: UploadJAR <session ID> <file>
```
Comment (loosebazooka) [inline kokoro/release_publish.sh]: Not sure why this method was here, so took it out.

**Comment by chanseokoh** [comment]
Did it not exit in some previous execution? With the following, it should exit anyway.
```
 # Fail on any error.
 set -e
```

**Comment by loosebazooka** [comment]
@chanseokoh that's a good point, let me go recheck the logs.

**Comment by loosebazooka** [comment]
Looks like the error was actually in the response? but curl succeeded, I'm going to recreate this locally if I can

**Comment by loosebazooka** [comment]
@chanseokoh yeah I saw that, it does work with 401's in my testing in our environments, my hope at least is that this improves things a little.

**Pull request merged**